File: Binary_search.c
#include<stdio.h>

int A[50];
int binary_search(int s, int f, int val);

int main(){
    int n, val, x, i;
    printf("enter the value of n: ");
    scanf("%d", &n);
    printf("enter the values in array: ");
    for(i=0; i<n; i++){
        scanf("%d", &A[i]);
      }
    printf("enter the value to be searched: ");
    scanf("%d", &val);
    x=binary_search(0, n-1, val);
    printf("%d", x);


    return 0;
}
int binary_search(int s, int f, int val){
    int m;
    if(s==f & A[s]!=val)
        return 0;
    if(s!=f){
    m=(f+s)/2;
    if(A[m]==val)
      return 1;
    if(val < A[m]){
        f=m-1;
        return binary_search(s, f, val);
    }
    if(val > A[m]){
        s=m+1;
        return binary_search(s, f, val);
      }
    }
}

File: python/Binary_search.py
def binarysearch(lst , x):
    l = 0
    u = len(lst)-1
    mid = 0
    while l<=u:
        mid = (l+u)//2
        if lst[mid] == x:
            return mid
        elif lst[mid] < x:
            l = mid+1
        else: #if lst[mid] > x
            u = mid-1
lst=[]
n=int(input("enter number of elements:"))
for i in range(n):
    ele=int(input("enter elements:"))
    lst.append(ele)
x = int(input("enter search element:"))
print("element present at index:",binarysearch(lst, x))
